Source/WebKit/WebProcess/Plugins/PDF/UnifiedPDF/UnifiedPDFPlugin.mm

```diff
@@ -2548,14 +2567,20 @@ static FloatRect computeMarqueeSelectionRect(const WebCore::FloatPoint& point1,
         if (!std::exchange(m_selectionTrackingData.lastHandledEventWasContextMenuEvent, false))
             return;
         break;
-    case ActiveStateChangeReason::SetCurrentSelection:
     case ActiveStateChangeReason::WindowActivityChanged:
+        if (!m_currentSelection || [m_currentSelection isEmpty])
+            return;
+        break;
+    case ActiveStateChangeReason::SetCurrentSelection:
         break;
     default:
         RELEASE_ASSERT_NOT_REACHED();
     }
 
-    m_contentsLayer->setNeedsDisplay();
+    auto selectionDocumentRectsToRepaint = selectionBoundsAcrossDocument(m_currentSelection);
```

Comment:
You should have a RetainPtr to m_currentSelection *on the stack* before passing it as parameter:
```
selectionBoundsAcrossDocument(RetainPtr { m_currentSelection }.get());
```
or introduce a member function named `protectedCurrentSelection()` which returns a `RetainPtr` and use it here:
```
selectionBoundsAcrossDocument(protectedCurrentSelection().get());
```

Reply:
Fixed in https://github.com/WebKit/WebKit/commit/9355bda2fca30c7dd4b198ad7569cb367250efcb